e2e/e2e.test.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @jest-environment node
+ */
+
+const { h } = require('@financial-times/x-engine') // required for <GreetingComponent>
+const { Serialiser, HydrationData } = require('@financial-times/x-interaction')
+const puppeteer = require('puppeteer')
+const ReactDOMServer = require('react-dom/server')
+import { GreetingComponent } from './common'
+const express = require('express')
+
+describe('x-interaction-e2e', () => {
+	let browser
+	let page
+	let app
+	let server
+
+	beforeAll(async () => {
+		app = express()
+		server = app.listen(3004)
+		app.use(express.static(__dirname))
+		browser = await puppeteer.launch()
+		page = await browser.newPage()
+	})
+
+	it('attaches the event listener to SSR components on hydration', async () => {
+		const ClientComponent = () => {
+			// main.js is the transpiled version of index.js, which contains the registered GreetingComponent, and invokes hydrate
+			return <script type="module" src="./main.js" charset="utf-8"></script>
+		}
+
+		const serialiser = new Serialiser()
+		const htmlString =
+			ReactDOMServer.renderToString(<GreetingComponent serialiser={serialiser} />) +
+			ReactDOMServer.renderToString(<HydrationData serialiser={serialiser} />) +
+			ReactDOMServer.renderToString(ClientComponent())
```

Comment:
**suggestion**: this could use a Fragment instead? i think it's more readable

```suggestion
		const htmlString = ReactDOMServer.renderToString(<>
			<GreetingComponent serialiser={serialiser} />
			<HydrationData serialiser={serialiser} />
			<ClientComponent />
		</>)
```